templates/cu_full/src/tasks.rs

```diff
@@ -66,18 +62,18 @@ impl CuTaskLifecycle for MyTask {
     // don't forget the other lifecycle methods if you need them: start, stop, preprocess, postprocess
 }
 
-impl CuTask for MyTask {
-    type Input = MyMsg;
-    type Output = MyMsg;
+impl<'cl> CuTask<'cl> for MyTask {
+    type Input = input_msg!(MyPayload);
```

Comment:
In order to remove the optional from user-space, im wondering if we'd be able to have the input msg like a macro as a struct, so

```rust
#[cu_input]
struct MyPayload {
  my_optional_value: Option<i32>,
  my_required_value: f32,
  my_array: Vec<u32>,
}
```

and the runtime would desugar these into payloads with optional values but only execute a CuTask if all its required inputs are available during runtime. This may better relate to optional outputs where a given node may or may not produce a value depending on the content of its inputs

Reply:
This is an interesting idea!
Basically we would add an option in the tree to say: all inputs mandatory
Then the runtime can basically serve a straight struct with no Option<>

But we would need to keep the non mandatory one, for example if you have a dual sensor path in your stack and one goes dark and doesn't produce any data, roboticians will want a way to continue in some degraded mode.

The other one is a super high frequency sensor merging with a slow one, you need to continue to feed the merger task.

Let's merge it like that and think about it again before the interface freeze at the beta stage. Let me add an issue to be sure we address it.

Thanks!